### DISCUSSION TITLE
Is there a way to clear MongoDB collection for a specific rate limiter programatically?

### DISCUSSION THREAD

**Discussion by yaroslav-perec** [animir/node-rate-limiter-flexible#159] [Q&A] [answered]
Hi! First of all, thank you for the great library! 

I have the next question "Is there a way to clear all data from a collection for a specific rate limiter"?

For example, I've created the next limiter:

```
const baseLimiter = new RateLimiterMongo({
    storeClient: connection,
    keyPrefix: 'rate_limiter_base',
    points: 100, // 100 attempts
    duration: 15 * 60, // within 15 minutes
});
```

Now I have this collection("rate_limiter_base") in my DB, and after some time I would like to clear all data from this collection using something like this:

`baseLimiter.deleteAll();`

But unfortunately, I couldn't find a similar API, so could you please help and tell me if there is a way to do this somehow?

**Comment by animir** [comment, marked as answer]
Hi, thank you.
There is no built-in method to clean up a collection.
There is [collection.deleteMany({})](https://docs.mongodb.com/drivers/node/current/usage-examples/deleteMany/) in mongodb package for that. For the `deleteMany` method you'd need a collection name, which can be got from baseLimiter like `baseLimiter.tableName`.

The code to delete all documents from collections:
```js
connection.collection(baseLimiter.tableName).deleteMany({})
```

Note, all expired documents removed from a collection automatically with expiring index built for every collection by `expire`.

**Reply by yaroslav-perec** [reply]
Thanks a lot! It works
